core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -854,7 +859,10 @@ public function copy($source=null, $destination=null)
 			throw new InternalServerErrorException('Attempt to copy the folder "' . $source . '" to "' . $strFolder . '" (circular reference).');
 		}
 
-		$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_ACTION_COPY, new DataContainerSubject($this->strTable, $source, array('destination' => $destination)));
+		$this->denyAccessUnlessGranted(
+			ContaoCorePermissions::DC_PREFIX . $this->strTable,
+			new CreateAction($this->strTable, array('id' => $source, 'pid' => $strFolder))
```

Comment:
shouldn't the ID be `$destination`? And should we also check for `ReadAction` on the source record?

Reply:
Done in https://github.com/contao/contao/pull/4770/commits/33551d83282ddcc61e1f56633b7ecaedb1a37e45#diff-f31b2d0b0120c3a32429a57b669102e7c9ef86a5ca152b2553602e44bd08b96d.